ng-devui-plus/smart-input/smart-input.component.ts

```diff
@@ -1,27 +1,47 @@
-import {Component,OnInit,Input,Output,EventEmitter,ElementRef,Renderer2,ViewChild} from '@angular/core';
+import {Component,OnInit,Input,Output,EventEmitter,ElementRef,Renderer2,ViewChild,forwardRef} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
   selector: 'd-smart-input',
   templateUrl: './smart-input.component.html',
-  styleUrls: ['./smart-input.component.scss']
+  styleUrls: ['./smart-input.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => SmartInputComponent),
+    multi: true
+}]
 })
-export class SmartInputComponent implements OnInit {
+export class SmartInputComponent implements ControlValueAccessor,OnInit {
   @Input() size: '' | 'sm' | 'lg';
   @Input() placeholder = "";
   @Input() disabled = false;
   @Input() clearable = false;
-  @Output() clearFunc = new EventEmitter<string>();
+  @Output() onClear = new EventEmitter<string>();
```

Comment:
```suggestion
  @Output() fieldClearEvent = new EventEmitter<string>();
```
Angular不推荐使用on作event的prefix，建议改下名字。